include/gridtools/storage/adapter/nanobind_adapter.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "../../common/array.hpp"
+#include "../../common/integral_constant.hpp"
+#include "../../sid/simple_ptr_holder.hpp"
+#include "../../sid/synthetic.hpp"
+#include "../../sid/unknown_kind.hpp"
+#include <algorithm>
+#include <nanobind/nanobind.h>
+#include <nanobind/ndarray.h>
+
+namespace gridtools {
+    namespace nanobind_sid_adapter_impl_ {
+        template <size_t, class>
+        struct kind {};
+
+        template <std::size_t UnitStrideDim = std::size_t(-1), class T, std::size_t... Sizes, class... Args>
+        auto as_sid(nanobind::ndarray<T, nanobind::shape<Sizes...>, Args...> ndarray) {
+            using sid::property;
```

Comment:
```suggestion
            static_assert(UnitStrideDim == std::size_t(-1), "Support for UnitStrideDim not implemented");
            using sid::property;
```
or actually implement support for it. The idea of the optimization is to mark the dimension which has unit stride and do a shift with the compile time known stride, possibly allowing more optimization for the compiler.

Or does maybe `nanobind::ndarray` expose the strides type and a possible compile time stride?

Reply:
`ndarray` doesn't seem to have such a construct. I added an option to specify any stride as static or dynamic.